controllers/update_pods.go

```diff
@@ -81,13 +82,35 @@ func (updatePods) reconcile(ctx context.Context, r *FoundationDBClusterReconcile
 func getPodsToUpdate(ctx context.Context, logger logr.Logger, reconciler *FoundationDBClusterReconciler, cluster *fdbv1beta2.FoundationDBCluster) (map[string][]*corev1.Pod, error) {
 	updates := make(map[string][]*corev1.Pod)
 
+	// Map of zones to the number of pods that are unavailable in that zone.
+	zoneToUnavailablePods := make(map[string]int)
```

Comment:
```suggestion
	faultDomainsWithUnavailablePods := make(map[string]fdbv1beta2{})
```

We don't need to count the Pods, as we don't care about the number of unavailable Pods but the number of fault domains containing at least one unavailable Pod.